contracts/PartyToken.sol

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.5;
+
+// ============ External Imports: Inherited Contracts ============
+import {ERC20VotesComp} from "@openzeppelin/contracts/token/ERC20/extensions/ERC20VotesComp.sol";
+import {ERC20Permit} from "@openzeppelin/contracts/token/ERC20/extensions/draft-ERC20Permit.sol";
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+/*
+Party Token
+by Anna Carroll
+*/
+contract PartyToken is ERC20VotesComp {
+    // ============ Immutables ============
+
+    address public immutable partyDAOMultisig;
+    address public immutable deprecationContract;
+
+    // ============ State ============
+
+    bool isUnlocked;
+
+    // ============ Events ============
+
+    event Unlocked();
+
+    // ======== Constructor =========
+
+    constructor(address _partyDAOMultisig, address _deprecationContract) ERC20("PartyDAO", "PARTY") ERC20Permit("PartyDAO") {
```

Comment:
wdyt about letting the multisig change the deprecationContract?

Reply:
this was likely already discussed -- but i'm worried about the fact that this is the exact same symbol and name as the existing PARTY token -- wish there was something different about them to distinguish

Reply:
🤔  i feel there really shouldn't be a scenario where we need to change the deprecation contract. since the token contract is staying with us forever i feel even more strongly about keeping the surface area streamlined.